### PR TITLE
fix: Provide single-quoted booleans to webpack.DefinePlugin.

### PR DESCRIPTION
**Pull request by becdot**
Boolean values must be provided to webpack's DefinePlugin  as single-quoted strings (e.g. `process.env.FALSEY_VALUE = 'false'`) instead of double-quoted strings (e.g. `process.env.FALSEY_VALUE = "'false'"` or `process.env.FALSEY_VALUE = JSON.stringify('false')`). This is because webpack evaluates double-quoted booleans as strings instead of booleans, resulting in unexpected behavior where `process.env.FALSEY_VALUE` gets evaluated as the string `"false"`, which is truthy.

The solution to this is to provide single-quoted booleans when creating a new `DefinePlugin` so that they can be evaluated correctly.

### CODE REVIEW

**Comment by mrsteele** [comment]
Hey @becdot and thanks for the PR!

I looked at it a bit, and please correct me if I am wrong but the general consensus seems to be that everything in `process.env` should be returned as a `String`.

Since this is the case, it can be assumed in your application that the properties in `process.env` should expect a string return, and if not set would return false due to js "falsy" rules.

I read up on this here: https://nodejs.org/api/process.html#process_process_env and specifically this sentence: `Assigning a property on process.env will implicitly convert the value to a string.`

**Comment by becdot** [comment]
Hi @mrsteele -- thanks for the response!  I don't think I did a very good good job explaining, so let me try again.  

You are totally correct -- it is also my understanding that `process.env.SOME_VALUE` should return a string.   If I were to use `webpack.DefinePlugin` on its own, I would do something like this:

```.js
// .env
FALSEY_VALUE=false
STRING_VALUE=foo

// index.js
if (process.env.FALSEY_VALUE) {
  console.log("This should not get executed");
} else {
  console.log("This should get executed");
}
// webpack.config.js
require('dotenv').config();
...
plugins: [
  new webpack.DefinePlugin({
    'process.env.FALSEY_VALUE': process.env.FALSEY_VALUE, // evaluated as "false"
    'process.env.STRING_VALUE': JSON.stringify(process.env.STRING_VALUE), // evaluated as "'foo'"
  })
]

// bundled code
if (false) {
  console.log("This should not get executed");
} else {
  console.log("This should get executed");
}
```
This will result in webpack inserting `false` and `"foo"` for `process.env.FALSEY_VALUE` and `process.env.STRING_VALUE`, respectively, into my bundled code.  All of this is as expected!

However, using `dotenv-webpack`, because all `process.env` values are passed to `JSON.stringify`, we get different behavior
```.js
// (dotenv src/index.js)
class Dotenv {
  constructor (options) {
    ...
    const formatData = Object.keys(vars).reduce((obj, key) => {
      // vars[key] is a string, because it's been loaded from the .env file
      obj[`process.env.${key}`] = JSON.stringify(vars[key])
      return obj
    }, {})
    // currently passes {"process.env.FALSEY_VALUE": "'false'", "process.env.STRING_VALUE": "'foo'"}
    return new DefinePlugin(formatData)
  }
}
```

This will result in webpack inserting `"false"` for `process.env.FALSEY_VALUE` into my bundled code, which will evaluate as truthy in my code!
```.js
// webpack.config.js
const Dotenv = require('dotenv-webpack');
 ...
plugins: [new Dotenv()]

// public/index.js
...
if (true) {
  console.log("This should not get executed");
} else {
  console.log("This should get executed");
}
```

I hope that helps explain the issue that I was experiencing, and why this made sense to me as a potential solution.  Looking forward to hearing your thoughts!

**Comment by mrsteele** [comment]
@becdot Yes I see what you are saying in proper context, thank you for providing that.

I would still like to maintain my argument that we should keep the strings in the code. I would also like to add that you might want to evaluate your personal use case to reverse your env variable usage.

For instance, if you are doing something like this (and sorry for the lame example, but its just off the top of my head):
```js
// SHOW_LOGS=false
if (process.env.SHOW_LOGS) {
  console.log('HA')
  // this would actually show, because `SHOW_LOGS`="false"
}
```

You can utilize this and simply not set the variables so that it is falsy as expected and conventional in `dotenv` and `node` (which applies directly to the use-case for this module).
```js
// SHOW_LOGS=
if (proccess.env.VERBOSE_LOGGING) {
  console.log('HA')
  // Its empty, which is false
}
```

If this was simply a `DefinePlugin` enhancement with no defined conventions or constitutions I think you would have an excellent case, but to remain consistent with hoe env variables work and the adopted conventions I think you should address your use of env variables.

Alternatively you could use explicit checks:

```js
process.env.TEST === "true"
process.env.TEST === "false"
// maybe use "1" and "0"
process.env.TEST === "1"
process.env.TEST === "0"
```

Let me know if you have anything else to add, but for now I'm closing this as I believe this module is in line with preexisting conventions.
